slam-libraries/viam-cartographer/src/slam_service/slam_service.h

```diff
@@ -143,6 +147,10 @@ class SLAMServiceImpl final : public SLAMService::Service {
 
     std::mutex map_builder_mutex;
     mapping::MapBuilder map_builder;
+
+    std::mutex pose_mutex;
```

Comment:
[opt] We might want to call this `response_mutex` or something else if we add `trajectory_nodes` here, but that can also wait until https://viam.atlassian.net/browse/DATA-706

Reply:
+1 on this, either now or in the future. I also like `slam_mutex` but might be not descriptive enough